effectful-co-log-core/src/Effectful/Colog/Core/Effect.hs

```diff
@@ -0,0 +1,67 @@
+module Effectful.Colog.Core.Effect
+  ( Log,
+    LogActionEff,
+    runLog,
+    logMsg,
+    logMsgs,
+    withLog,
+  )
+where
+
+import Colog.Core (LogAction (..))
+import qualified Colog.Core as Colog
+import Data.Foldable (traverse_)
+import Effectful.Dispatch.Static
+  ( StaticRep,
+    evalStaticRep,
+    forkEnv,
+    getEnv,
+    localStaticRep,
+    unEff,
+    unsafeEff,
+    unsafeEff_,
+  )
+import Effectful.Monad
+  ( Dispatch (Static),
+    DispatchOf,
+    Eff,
+    Effect,
+    type (:>),
+  )
+
+-- | An effect for composable, contravariant and comonadic logging using the @co-log@ library.
+data Log msg :: Effect
+
+type instance DispatchOf (Log msg) = 'Static
+
+newtype instance StaticRep (Log msg) = Log (LogAction IO msg)
+
+-- | 'LogAction' that works directly with the 'Eff' monad.
+type LogActionEff es msg = LogAction (Eff es) msg
```

Comment:
This should always be `LogAction IO msg`, having it use `Eff` introduces a lot of complexity for no particular gain IIUC.

Reply:
This implementation enables `LogAction`s like [`logMessagePure`](https://github.com/ambroslins/effectful-contrib/blob/fa858d1927e2ad4d0a88adfa893864800488799b/effectful-co-log/src/Effectful/Colog/Pure.hs#L23) without running them in `IO`.

I know pure logging is probably hardly used. However IMHO it can be useful to specify the Effects a `LogAction` requires.
E.g. the [`defaultFieldMap`](https://hackage.haskell.org/package/co-log-0.4.0.1/docs/Colog-Message.html#v:defaultFieldMap) could only require the `Time` and `Concurrent` Effects (if I figure out how to convert `UTCTime` to the chronos `Time`) instead of `IOE`. Also with a background logger my `LogAction` would only require the `Concurrent` Effect, so I know that this will not run arbitrary IO in the current thread. 

These are probably also no the most common use cases, but I think this is the main difference to packages like `effectful-log-base`.

Reply:
I'd say that in this case it's better to make it dynamically dispatched.